physics/processes/inc/pair_production.h

```diff
@@ -25,26 +25,37 @@ __device__ void pair_production::GenerateInteraction(int particle_index, adept::
 {
   track *mytrack = &((*block)[particle_index]);
 
-  float eloss = 0.5f * mytrack->energy;
+  float esecond = 0.5f * mytrack->energy;
 
   // pair production
-  mytrack->energy -= eloss;
+  mytrack->energy -= esecond;
   mytrack->energy_loss           = 0;
-  mytrack->number_of_secondaries = 1;
+  mytrack->number_of_secondaries++;
```

Comment:
This might be badly named, but `number_of_secondaries = 1` is actually correct I think: this sets the number of secondaries produced in this process. This is accumulated in the global scoring which is now overcounting (ie a second secondary produced by the same primary track now counts as two additional particles, then three and so on)

Reply:
As it is a member of the track, it must mean the secondaries produced by the track.  In this sense I can see uses.
It is not clear to me what use a flag that a secondary was produced in the current step would have.

Reply:
It's used to count how many secondaries have been produced globally. Pair production in the current step adds exactly one particle, not more versions of all particles ever produced. (As I said, maybe badly named and other schemes potentially make more sense, but that's what the current code does)

Reply:
Can you please revert this change for now? As I said, it gives wrong counts with the current code.

Reply:
It's like Jonas says, the number of secondaries created by the given process. The same is for energy loss. I agree, that this should go into some 'Geant4-like particle change' struct in the long term. We need to figure out how we want the process to communicate back what has changed to the track.

Reply:
Yes, we should change it, but we shouldn't break the current examples in this PR.

> If it's supposed to flag whether the track has created more than zero secondaries, the current 'naive' scoring can simply check whether the 'true' number_of_secondaries is above zero

Unfortunately no, because that would assume a process creates secondaries in every step after it has created at least one...

Reply:
This is now be fixed - the output of example2 is reproduced as before.